### PR TITLE
feat(docs): ThemeCreator

### PR DESCRIPTION
**Pull request by tripleWdotcom**
With Theme-Creator you can generate a `THEME`  file with custom colors by changing the color of the 'slots'.
(All the colors from `Palette` are eligible)

Change a 'slot' color by clicking on the color segment and choosing a color from the Color-Picker.

https://user-images.githubusercontent.com/67017215/122547654-ac244d80-d030-11eb-9a44-42e330ca5c99.mp4

When the Preview has the desired look, download the file.

https://user-images.githubusercontent.com/67017215/122547770-d1b15700-d030-11eb-8b56-cbbf2395c012.mp4

Then just paste the `file` in the project and `import` it from its respective `path`, 
Example:
```tsx
 import { myCustomTheme } from "./pathTo<downloadedFile>"

        <PracticalProvider>
          <ThemeProvider theme={myCustomTheme}>
            <App />
          </ThemeProvider>
        </PracticalProvider>
```

https://user-images.githubusercontent.com/67017215/122548198-5308e980-d031-11eb-93d0-3d88e6935cb7.mp4

### CODE REVIEW

File: packages/docs/src/mdx/colorSelector/ColorPicker.tsx [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+  const colorSegment = useMemo(() => palette[value](), [value])
+
+  const checkDirection = useCallback(() => {
+    const { clientHeight } = document.documentElement
+    const elem = anchorEl?.getBoundingClientRect()
+    if (elem) {
+      const diff = Math.floor(clientHeight - elem.bottom)
+
+      // 300px is the height of <PaletteContainer>
+      if (diff < 300) setDirectionUp(true)
```
Comment (steabert) [inline packages/docs/src/mdx/colorSelector/ColorPicker.tsx:111]: I think it would be better to use `300` as a constant in this file in that case.

Comment (tripleWdotcom) [reply packages/docs/src/mdx/colorSelector/ColorPicker.tsx]: Thanks for the feedback, It is changed now.

**Review by steabert — APPROVED** [review]
LGTM

**Comment by Tigge** [comment]
This is done, but needs to be renamed to "Theme creator" instead of color selector. So if you could update the names both in the code, commit message and this PR we could get this merged. Thanks!

**Pull request merged**